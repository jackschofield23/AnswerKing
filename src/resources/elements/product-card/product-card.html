<template>
  <div class="card">
    <header class="card-header">
      <p class="card-header-title">
        ${product.name} | ${product.category.name}
      </p>
    </header>
    <div class="card-content">
      <div class="content">
        <!-- display description and price -->
        <!-- price needs a value converter -->
      </div>
    </div>
    <footer class="card-footer">
      <a class="card-footer-item">Add to Order</a>
    </footer>
  </div>
</template>
