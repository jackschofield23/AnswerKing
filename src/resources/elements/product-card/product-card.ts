import { bindable } from 'aurelia-framework';

export class ProductCardCustomElement {
  @bindable public product: IProduct;
}
