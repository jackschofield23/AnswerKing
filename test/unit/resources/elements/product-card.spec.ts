import { bootstrap } from 'aurelia-bootstrapper';
import { PLATFORM } from 'aurelia-framework';
import { StageComponent, ComponentTester } from 'aurelia-testing';

import { ProductCardCustomElement } from 'resources/elements/product-card/product-card';

describe('Product Card Element', () => {
  let component: ComponentTester<ProductCardCustomElement>;
  const product: IProduct = {
    id: '89828e46-6cff-438f-be1a-6fa9355cfe24',
    name: 'Chips',
    description: 'Nothing else to add.',
    price: 2.99,
    category: {
      id: 'e32701f8-d644-4d5d-bd52-2a31fdaba3df',
      name: 'Sundries',
      description: ' Things that go with things.',
    },
    retired: false,
  };

  beforeEach(() => {
    component = StageComponent.withResources([
      PLATFORM.moduleName(
        '../../../src/resources/elements/product-card/product-card'
      ),
      // add currency value converter here
    ])
      .inView('<product-card product.bind="product"></product-card>')
      .boundTo({
        product,
      });
  });

  afterEach(() => component.dispose());

  it('renders the correct information', async () => {
    await component.create(bootstrap);
    expect(component.element).toMatchSnapshot();
  });
});
