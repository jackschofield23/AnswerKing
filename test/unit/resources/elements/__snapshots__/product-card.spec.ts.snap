// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Element renders the correct information 1`] = `
<product-card
  au-target-id="4"
  class="au-target"
  product.bind="product"
>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title"
      >
        
        Chips | Sundries
      
      </p>
    </header>
    <div
      class="card-content"
    >
      <div
        class="content"
      >
        <p>
          Nothing else to add.
        </p>
        <p>
          Â£2.99
        </p>
      </div>
    </div>
    <footer
      class="card-footer"
    >
      <a
        class="card-footer-item"
      >
        Add to Order
      </a>
    </footer>
  </div>
</product-card>
`;
